### PR TITLE
Rename repo from "nupic.cpp" to "htm.core"

### LINKED ISSUE #524: Rename the repo to ???
With the ongoing changes (#518 ) to uncouple htm from the NuPIC (tm), we should rename the repo. 

What is the agreed name? 
- htm.cpp ? 
- htm + something with community ?

### PR DESCRIPTION
**Pull request by ctrl-z-9000-times**
Fixes #524 

This probably won't pass CI until the repo is renamed to "htm.core".

### CODE REVIEW

File: bindings/py/packaging/setup.py [hunk 1]
```diff
@@ -238,7 +238,7 @@ def generateExtensions():
     #     https://opensourceforu.com/2010/OS/extending-python-via-shared-libraries
     #     https://docs.python.org/3/library/ctypes.html
     #     https://docs.python.org/2/library/imp.html
-    name="nupic.bindings",
+    name="htm.core",
```
Comment (dkeeney) [inline bindings/py/packaging/setup.py]: I don't think this is the repo name.  Its the name given to the package but yes, htm.core is a better name.

**Comment by breznak** [comment]
> When we rename the repo we should create a dummy repro named nupic.cpp which is a pointer to the new repo name.

good point, and Github does just that. See link below, as this repo used to be a community/nupic.core (forked) 

https://github.com/htm-community/nupic.core 

I'm going ahead and renaming the repo now..

**Comment by breznak** [comment]
New times ahead for htm.core repo! :)

**Comment by dkeeney** [comment]
Thanks for doing the repo renaming.

**Comment by ctrl-z-9000-times** [comment]
I think that the CI might be broken?

Also, I noticed that the CI indicators on the README should be updated since they contain the repo name inside of the URL.

**Comment by breznak** [comment]
> I think that the CI might be broken? 

the CI itself is running ok, its the webhooks that did not handle the rename. Problem is I reviewed the settings here in GH and under Webhooks did not find anything repo-name related. 

https://travis-ci.org/htm-community/htm.core/pull_requests

**Comment by breznak** [comment]
Fixed. 

For CircleCI: 
I toggled "Monitor these repos:" where I selected the htm.core (off, on again) in Circle's settings. 

For Appveyor: 
In its settings was URL to nupic.cpp, changed that to htm.core.

**Comment by ctrl-z-9000-times** [comment]
I fixed the CI badges in the README, this should be ready to merge once it passes CI.

Thank you Breznak for renaming the repo and fixing the CI hooks!

**Pull request merged**